client/src/app/controllers/search-project-controller/search-project.service.ts

```diff
@@ -0,0 +1,89 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { AuthService } from '../auth-controller/auth.service';
+import { environment } from 'src/environments/environment';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class SearchProjectService {
+  apiUrl = environment.apiUrl;
+
+  constructor(private http: HttpClient, private authService: AuthService) { }
+  //This method is pretty simple, just fill in the fields that exist or will be used with the search or null if not. To the best of my knowledge, js/ts does 
+  //not support named parameters so if you wanted to only serach with deadline, the correct call to this method would be searchProjectsMultipleParams(null, null, null, null, null, null, DATE PARAMETER)
+  searchProjectsMultipleParams(query: string | null, majors: string[] | null, GPA: number | null, npp: number | null,
+    pageNum: number | null, posted: Date | null, deadline: Date | null): Observable<any> {
+    const headers = this.authService.getHeaders();
+    let request = "?";
+    let useAndSym = false;
+
+    if (query) {
+      request += "query=" + query;
+      useAndSym = true;
+    }
```

Comment:
Also, I think this method could be refactored to be a couple of lines using Angular's `HttpParams`. See [this post](https://stackoverflow.com/a/58622743) for some good info on how to do that.

Reply:
I checked that post, isn't it nearly the same as the way I had done it?

Reply:
This was difficulty to implement and considering that it still works with 60 lines, we are going to fix it later :clown: